app/jobs/v3/delete_route_binding_job.rb

```diff
@@ -47,24 +49,43 @@ def perform
         if @first_time
           @first_time = false
           delete_result = action.delete(binding, async_allowed: true)
-          return finish if delete_result.is_a? V3::ServiceRouteBindingDelete::DeleteComplete
+          if delete_result[:finished]
+            finish
```

Comment:
If you don't return here the function will proceed (therefore executing the polling on line 57). Is this intentional?

Reply:
hmm - I thought that was returning. There is a test [here](https://github.com/cloudfoundry/cloud_controller_ng/blob/3bea98c52a0680498b9bfadcc683749df0a555a5/spec/unit/jobs/v3/delete_route_binding_job_spec.rb#L54-L71) checks that. 
Does that look right?

Reply:
Hum... that test looks wrong 🙈

It's probably passing because we never call `subjet.perform` in the second `it` block

Reply:
you're right - surpriseee
I fixed it